spi/src/main/java/org/eclipse/dataspaceconnector/spi/contract/ContractOfferFrameworkQuery.java

```diff
@@ -14,29 +14,28 @@
 
 package org.eclipse.dataspaceconnector.spi.contract;
 
-import java.security.Principal;
+import org.eclipse.dataspaceconnector.spi.iam.VerificationResult;
 
 /**
  * The {@link ContractOfferFrameworkQuery} narrows down the number of queried {@link ContractOfferTemplate}.
  */
 public class ContractOfferFrameworkQuery {
 
-    // TODO define who the principal is and rename it (promisee, assigne, recipient, daps token, etc.?)
-    private Principal principal;
+    private VerificationResult verificationResult;
```

Comment:
feels strange to have a VerificationResult as a field. I think it should just be used as a return value.

Reply:
Yea. We intent here to enable the ContractOfferFramework to be enable to create custom offers for each request. But I think you are right, maybe it's not a good idea to pass the whole `VerificationResult` object here.

Would it resolve the issue if we extract the claims from the `VerificationResult` and wrap it in another object? Maybe a principal?